docs/rules/require-prop-types.md

```diff
@@ -34,6 +34,20 @@ props: {
 }
 ```
 
+```js
+props: {
+  status: {
+    required: true,
+    validator: function (value) {
```

Comment:
Or actually, perhaps we can remove this code, and use above example with `validate` property changed to proper `validator`?

Reply:
I didn't feel right editing another example at the time. Now I looked up vue code and found no mention of `validate` as prop option at all. So it makes sense to change it to `validator`. However, I think it would be clearer if the docs contained at least three cases:
```js
// Without options, just type reference
props: {
  status: String
}

// With options with type field
props: {
  status: {
    type: String,
    required: true
  }
}

// With options without type field but with validator field
props: {
  status: {
    required: true,
    validator: function (value) {
      return (
        value === null ||
        Array.isArray(value) && value.length > 0
      )
    }
  }
}
```
so the readers understand that `type` is not required if `validator` is present. There is also the case when both `type` and `validator` are used but I don't think it's relevant here since both are covered separately.

So what if I remove `validate` from the second example, so it's just about `type`, and also add explicit comments like above?

Reply:
Sounds good to me @st-sloth , go for it :)